src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOQuerySupport.scala

```diff
@@ -96,10 +95,10 @@ trait ElasticSearchDAOQuerySupport extends ElasticSearchDAOSupport {
   def buildAggregateQueries(client: TransportClient, indexname: String, criteria: LibrarySearchParams): Seq[SearchRequestBuilder] = {
     // for aggregations fields that are part of the current search criteria, we need to do a separate
     // aggregate request *without* that term in the search criteria
-    (criteria.fieldAggregations intersect criteria.filters.keySet.toSeq) map { field =>
+    (criteria.fieldAggregations.keySet.toSeq intersect criteria.filters.keySet.toSeq) map { field: String =>
       val query = createQuery(criteria.copy(filters = criteria.filters - field))
       // setting size to 0, we will ignore the actual search results
-      addAggregationsToQuery(createESSearchRequest(client, indexname, query, 0, 0), Seq(field), criteria.maxAggregations)
+      addAggregationsToQuery(createESSearchRequest(client, indexname, query, 0, 0), criteria.fieldAggregations.filterKeys(key => key == field))
```

Comment:
avoid using `filterKeys` (and `mapValues`) in Scala - it has bizarre and unpredictable side effects. See https://issues.scala-lang.org/browse/SI-4776 for the root issue, and http://stackoverflow.com/questions/14882642/scala-why-mapvalues-produces-a-view-and-is-there-any-stable-alternatives, and google around. This particular usage is probably safe, but as a rule we want to eliminate it from our codebase. You can use plain `filter` instead, even though it requires more code.